### DISCUSSION TITLE
Refresh Token only working once?

### DISCUSSION THREAD

**Discussion by wischnu** [bimmerconnected/bimmer_connected#678] [General] [answered]
Hello,
I have changed my python script to make use of the Captchas and the Refresh Tokens. One script is logging in to BMW and is saving the Refresh Token and the GCID, the second one is using it to perform different actions.

So far it is working, however, after only **one** action using the Refresh Token (e.g. executing vehicle.remote_services.trigger_remote_air_conditioning()) it has expired and I need to login again using a new Captcha. 

Is this normal?

**Comment by rikroe** [comment, marked as answer]
Does https://github.com/bimmerconnected/bimmer_connected/discussions/677#discussioncomment-11357248 help you?

**Comment by wischnu** [comment]
Yes thanks. This clarified it.
